### PR TITLE
Add Terraform support for GKE enable_nested_virtualization option

### PR DESCRIPTION
**Pull request by lornaluo**
This PR adds the `advanced_machine_features.enable_nested_virtualization` field in `node_config`. It enables the user to use terraform to config [nested VMs with GKE clusters](https://cloud.google.com/kubernetes-engine/docs/how-to/nested-virtualization).

 I acknowledge that I have:
* [x]   Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description].
* [x]   Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
* [x]   [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
* [x]   [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
* [x]   Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note: enhancement 
container: added `enable_nested_virtualization` field to `google_container_node_pool` resource.
```

### CODE REVIEW

**Review by shuyama1 — COMMENTED** [review]
Looks like the field just got released today in the [latest version v0.177.0](https://pkg.go.dev/google.golang.org/api@v0.177.0/container/v1beta1#AdvancedMachineFeatures). You'll need to bump `google.golang.org/api` to the latest version in mmv1/third_party/terraform/go.mod.erb

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb [hunk 1]
```diff
@@ -2138,6 +2138,69 @@ resource "google_container_node_pool" "np" {
...
+  deletion_protection = false
+  network    = "%s"
+  subnetwork    = "%s"
+
+  node_config {
+    machine_type = "c2-standard-4"
+	advanced_machine_features {
+    threads_per_core = 1
+    enable_nested_virtualization = "%t"
+	}
```
[the remark this answers is not in the document]
Comment (lornaluo) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb]: Thanks for the suggstion. Updated.

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb [hunk 2]
```diff
@@ -2138,6 +2138,69 @@ resource "google_container_node_pool" "np" {
...
+  cluster            = google_container_cluster.cluster.name
+  initial_node_count = 2
+
+  node_config {
+    machine_type = "c2-standard-4"
+	advanced_machine_features {
+    threads_per_core = 1
+    enable_nested_virtualization = "%t"
+	}
+  }
```
[the remark this answers is not in the document]
Comment (lornaluo) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb]: Done

**Review by shuyama1 — COMMENTED** [review]
Mostly LGTM!. We have `google_container_cluster` and `google_container_node_pool` sharing the `node_config` schema. Would you mind also adding a test coverage in `google_container_cluster` tests according to the [missing test report](https://github.com/GoogleCloudPlatform/magic-modules/pull/10558#issuecomment-2088835350). You can just simply modify an existing test to include the field. Thanks!

**Review by shuyama1 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**